pre_commit/runner.py

```diff
@@ -43,17 +44,39 @@ def repositories(self):
             repository.require_installed()
         return repositories
 
+    def get_hook_path(self, hook_type):
+        return os.path.join(self.git_root, '.git', 'hooks', hook_type)
+
     @cached_property
     def pre_commit_path(self):
-        return os.path.join(self.git_root, '.git', 'hooks', 'pre-commit')
+        return self.get_hook_path('pre-commit')
+
+    @cached_property
+    def pre_push_path(self):
+        return self.get_hook_path('pre-push')
+
+    @cached_property
+    def pre_template(self):
+        return resource_filename('hook-tmpl')
 
     @cached_property
+    def pre_push_template(self):
+        return resource_filename('pre-push-tmpl')
```

Comment:
I appreciate your pursuit of the code, especially for the requirement of coverage. But I think you with a lot of personal subjective for ths project. I open the [pre-push](https://github.com/dongweiming/pre-push) is because I feel your can hard accepte others opinion.

These cost me a lot of time. i think you can merge it and modify somethings you want. or you can do this part by yourself. Thank you so much for review!